### PR TITLE
DOC/TST: Document numpy 2.0 support and add tests for string array

### LINKED ISSUE #58104: COMPAT/TST: Ensure that numpy 2 string dtype converts to object array in pandas 2.2.x
We should add a test for this, given that numpy 2.0 support is happening for pandas 2.2.2.

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #58104  (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -3052,6 +3053,24 @@ def test_from_dict_with_columns_na_scalar(self):
...
+    @pytest.mark.parametrize(
+        "data",
+        [
+            {"a": ["a", "b", "c"], "b": [1.0, 2.0, 3.0], "c": ["d", "e", "f"]},
+        ],
+    )
+    def test_np_string_array_object_cast(self, data):
+        from numpy.dtypes import StringDType
+
+        data["a"] = np.array(data["a"], dtype=StringDType())
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:3069]: Curious what happens when `dtype=StringDType()` is passed to pandas `Series` or `DataFrame` constructor? Fine if it doesn't work/converts to object but would be good to note if something unexpected happens

**Review by mroeschke — APPROVED** [review]

File: pandas/tests/frame/test_constructors.py (hunk 1 above)
Comment (lithomas1) [reply pandas/tests/frame/test_constructors.py:3069]: I think it silently casts to object.

FWIW, I think the numpy fixed length strings also do this.

**Comment by lorentzenchr** [comment]
I don't know where to ask: Is there a ETA for pandas 2.2.2 with numpy 2.0(rc) support?

**Review by WillAyd — APPROVED** [review]

**Comment by lithomas1** [comment]
> I don't know where to ask: Is there a ETA for pandas 2.2.2 with numpy 2.0(rc) support?

Release is expected to happen today.

**Pull request merged**